torch_ttnn/passes/memory_pass.py

```diff
@@ -0,0 +1,240 @@
+import torch
+import ttnn
+from torch.fx.passes.infra.pass_base import PassBase, PassResult
+from torch_ttnn.passes.lowering.add_data_move_pass import is_tt_compute
+
+from torch_ttnn.mem_utils import *
+
+
+def memory_footprint(gm: torch.fx.GraphModule) -> torch.fx.GraphModule:
+    mm = MemoryManager(device_name, cores, L1_mem, circular_buffer)
+
+    op_registry = OpRegistry()
+    mm.logs += "Track memory footprint for each of the ops:\n"
+    print("Track memory footprint for each of the ops:")
+    nodes = list(gm.graph.nodes)
+
+    # Tensor ID allocation tracker
+    last_assigned_tid = 1
+    # Memory address allocation tracker
+    last_assigned_addr = 0
+    # List of already executed nodes on device
+    computed_nodes = []
+    # Data points containing info of on device tensors allocation & deallocation for ttnn ops
+    # Useful for visualization by plotting in a chart
+    data_points = {}
+    # On device tensors info
+    on_device_meta = []
+
+    # Tensor IDs allocation for ttnn ops input & output
+    tt_compute_count = 0
+    for node in nodes:
+        if is_tt_compute(node):
+            tt_compute_count += 1
+            # For inputs
+            for input_node in node.all_input_nodes:
+                if input_node not in mm.node_to_tid_map:
+                    mm.node_to_tid_map[input_node] = last_assigned_tid
+                    last_assigned_tid += 1
+            # For output
+            if node not in mm.node_to_tid_map:
+                mm.node_to_tid_map[node] = last_assigned_tid
+                last_assigned_tid += 1
+
+    # print(f"Tensor IDs for nodes on device:")
+    # for key, value in mm.node_to_tid_map.items():
+    #     print(f"Node: {key} - {value}")
+
+    # Tracks the total compute memory of the model
+    mm.logs += f"Total tt compute nodes: {tt_compute_count}\n"
+    print(f"Total tt compute nodes: {tt_compute_count}")
+    i = 0
+    overflow_ops = []
+
+    for node in nodes:
+        on_device = False
+        # Has to be a ttnn op
+        if is_tt_compute(node):
+            mm.logs += "---------------------------------------------------\n"
+            print("\n---------------------------------------------------")
+            mm.logs += f"tt compute node numbered: {i}\n"
+            print(f"tt compute node numbered: {i}")
```

Comment:
mm, maybe 
```
mm.log(message)
```
+

```
def log(mesage):
  messages += message;
  print(message);
```

Reply:
please calibrate your internal compass on the code that smells

Reply:
Got it.